packages/services/model/src/modules/models/bkt.ts

```diff
@@ -0,0 +1,69 @@
+//import { ModelState, Content, KC, Schemas } from "api-base";
+
+import { ModelState, Action, KC, Schemas } from "api-base";
+import parameters from "../parameters.json";
+
+export const bkt = (
+  D: string[],
+  M: ModelState | null,
+  A: (Action & {
+    kcs: KC[];
+  })[]
+) => {
+  const params = Schemas.parametersJson.parse(parameters); //params of json
+
+  const newMJSON = //json of the last model or create
+    M?.json ??
+    D.reduce(
+      (a, v) => ({
+        ...a,
+        [v]: {
+          level:
+            params[v]?.known ??
+            params["default"]?.known ??
+            parameters.default.known ??
+            0.1,
+          mth: params[v]?.mth ?? parameters.default.mth,
+        },
+      }),
+      {}
+    ); //si ya existe el nuevo json es el json anterior y se actualizan los kcs utilizados, si no se crea un JSON con valores vacios
+
+  let newModelJSON = Schemas.newModel.parse(newMJSON); //json del modelo que es retornado con BKT aplicado
+
+  A.map((act) => {
+    //array of action from oldest to most recent (.slice(0).reverse() innecesary)
+    act.kcs.map((kc) => {
```

Comment:
`.map` es para retornar un arreglo a partir de otro arreglo, usa `for (const act of A)` en vez de `.map`